src/main/kotlin/onboarding/Problem1.kt

```diff
@@ -1,5 +1,61 @@
 package onboarding
 
+import kotlin.math.max
+
 fun solution1(pobi: List<Int>, crong: List<Int>): Int {
-    TODO("프로그램 구현")
+    if (validateTwoPages(pobi[0], pobi[1]) && validateTwoPages(crong[0], crong[1])) {
+        val pobiMaxNumber = calcMaxPages(pobi)
+        val crongMaxNumber = calcMaxPages(crong)
+        return compareNumber(pobiMaxNumber, crongMaxNumber)
+    }
+    return -1
+}
+
+private fun validateTwoPages(leftNumber: Int, rightNumber: Int): Boolean {
+    val zero = 0
+    val one = 1
+    val two = 2
+    val minimumPage = 1
+    val maximumPage = 400
+
+    when {
+        leftNumber !in (minimumPage..maximumPage) -> return false
+        rightNumber !in (minimumPage..maximumPage) -> return false
+        leftNumber % two == zero -> return false
+        rightNumber % two == one -> return false
+    }
+    return (leftNumber + one) == rightNumber
 }
+
+private fun calcMaxPages(pages: List<Int>): Int {
+    var maxValue = 0
+
+    pages.forEach { pageNumber ->
+        maxValue = max(maxValue, getAddedDigitSum(pageNumber))
+        maxValue = max(maxValue, getMultipliedSum(pageNumber))
+    }
+    return maxValue
+}
+
+private fun getAddedDigitSum(number: Int): Int =
+    number.toString()
+        .fold(0) { total, digit -> total + Character.getNumericValue(digit) }
+
+private fun getMultipliedSum(number: Int): Int =
+    number.toString()
+        .fold(1) { total, digit -> total * Character.getNumericValue(digit) }
+
+private fun compareNumber(number1: Int, number2: Int): Int {
```

Comment:
```suggestion
private fun compareNumber(pobiMaxNumber: Int, crongMaxNumber: Int): Int {
```
인자로 넘어오는 변수명을 디테일하게 적으시거나 diff 자체를 인수로 받으셔도 될 것 같아요!
```
private fun compareNumber(diff: Int): Int 
```

불변으로 사용되는 상수들의 경우에는 대문자를 사용하는 스네이크케이스를 사용한다고 하던데

```
    val pobiWinningNumber = 1
    val crongWinningNumber = 2

    val POBI_WINNING_NUMBER = 1
    val CRONG_WINNING_NUMBER = 2
```

이거를 함수내에서 잠시 사용되는 변수의 네이밍에도 적용해야할지는 모르겠네요..?!

어쨌든 좋은 소스 잘보고 갑니닷!

Reply:
코드리뷰 감사합니다!!

compare할때는 아무런 숫자나 받아도 된다고 생각해서 number1, 2와 같은 네이밍을 했는데
읽는 사람 입장에서는 조금 헷갈릴 수 있겠군요..ㅠㅠ

diff로 받는 것도 좋은 방법인 것 같아요! ㅎㅎ
또는 변수명을 number1, 2라는 것보다 pobiMaxNumber와 같은 구체적인 네이밍을 하는 것이 가독성에 유익할 것 같네요.


<br>

   ```
val pobiWinningNumber = 1
val crongWinningNumber = 2

val POBI_WINNING_NUMBER = 1
val CRONG_WINNING_NUMBER = 2
```
    
이 부분은 찾아봐도 잘 안 나와서 상수임에도 카멜케이스를 사용했어요 ㅎㅎ!
저도 함수내에서 잠시 사용되는 경우에도 적용해야 하는지는 잘 모르겠습니다..ㅠㅠ
만약 프로젝트 단위의 규모라면 따로 Constants.kt 파일을 만들어서 스네이크 케이스를 적용하고 사용하는 것도 좋을 것 같아요!

리뷰 감사합니다!!